### DISCUSSION TITLE
Can't add a stream - stream always offline

### DISCUSSION THREAD

**Discussion by OpenK54** [ZoeyVid/NPMplus#435] [Q&A] [answered]
I'am not able to a stream anymore. It worked flawlessly but not anymore. It says invalid port in 0.0.0.0.0.0.0.0:443. I don't understand this because the adress is not an ip adress. Can someone help?

**Comment by Zoey2936** [comment]
Can you please show me your conf?

**Comment by OpenK54** [comment]
Sure:

# ------------------------------------------------------------
# 50832 TCP: 0 UDP: 1
# ------------------------------------------------------------


server {
    listen 0.0.0.0:50832 udp;
    listen [::]:50832 udp;

    proxy_pass 192.168.0.46:51832;

    # Custom
    include /data/nginx/custom/server_stream[.]conf;
    include /data/nginx/custom/server_stream_udp[.]conf;
}

Thanks for your support!

Cheers, Rafael

Von: Zoey ***@***.***>
Gesendet: Donnerstag, 12. Oktober 2023 08:08
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


Can you please show me your conf?

-

**Reply by Zoey2936** [reply]
can you send me your docker logs, please?

**Comment by OpenK54** [comment]
Sure (I replaced the domain name). Thank you!

Von: Zoey ***@***.***>
Gesendet: Donnerstag, 12. Oktober 2023 21:34
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


can you send me your docker logs, please?

-
Reply to this email directly, view it on GitHub<https://github.com/ZoeyVid/NPMplus/discussions/435#discussioncomment-7267096>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7WJZYUXIC2MZXLNSY6N7K3X7BA3JANCNFSM6AAAAAA54VLICU>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

      '8o34/m8Fxw==\r\n' +
      '-----END CERTIFICATE-----\r\n'
  },
  modified_on: '2023-10-12 19:44:46',
  created_on: '2023-10-12 19:44:46',
  owner_user_id: 1,
  provider: 'other',
  nice_name: 'Rapid'
}
Duplicate relation "access_list" in a relation expression. You should use "a.[b, c]" instead of "[a.b, a.c]". This will cause an error in objection 2.0
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/proxy_host/1.conf"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 19:46:06 [warn] 5977#5977: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:46:06 [warn] 5977#5977: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:46:06 [warn] 5977#5977: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:46:06 [notice] 5977#5977: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
2023/10/12 19:51:10 [error] 5982#5982: *51 connect() failed (111: Connection refused) while connecting to upstream, client: 178.197.215.72, server: XXXX, request: "GET / HTTP/2.0", upstream: "https://192.168.0.11:55555/", host: "XXXXX:8950"
2023/10/12 19:51:10 [error] 5982#5982: *51 connect() failed (111: Connection refused) while connecting to upstream, client: 178.197.215.72, server: XXXXX, request: "GET /favicon.ico HTTP/2.0", upstream: "https://192.168.0.11:55555/favicon.ico", host: "XXXX:8950", referrer: "XXXXX:8950/"
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 19:53:39 [warn] 6272#6272: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:53:39 [warn] 6272#6272: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:53:39 [warn] 6272#6272: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:53:39 [notice] 6272#6272: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf.err
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf.err"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 19:54:08 [warn] 6438#6438: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:08 [warn] 6438#6438: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:08 [warn] 6438#6438: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:08 [notice] 6438#6438: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 19:54:15 [warn] 6585#6585: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:15 [warn] 6585#6585: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:15 [warn] 6585#6585: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:15 [notice] 6585#6585: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/6.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/6.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/6.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/6.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/6.conf.err
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/6.conf.err"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 19:54:42 [warn] 6753#6753: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:42 [warn] 6753#6753: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:42 [warn] 6753#6753: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:54:42 [notice] 6753#6753: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)

**Comment by OpenK54** [comment]
Log from another server (same problem). Cheers, Rafael

Von: Zoey ***@***.***>
Gesendet: Donnerstag, 12. Oktober 2023 21:34
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


can you send me your docker logs, please?

-
Reply to this email directly, view it on GitHub<https://github.com/ZoeyVid/NPMplus/discussions/435#discussioncomment-7267096>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A7WJZYUXIC2MZXLNSY6N7K3X7BA3JANCNFSM6AAAAAA54VLICU>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

Date:     Thu Oct 12 14:14:45 UTC 2023
User:     root
PUID:     0
User ID:  0
PGID:     0
Group ID: 0
-------------------------------------
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
[Global   ] › ℹ  info      Using Sqlite: /data/etc/npm/database.sqlite
[Migrate  ] › ℹ  info      Current database version: none
[IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
[IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
[SSL      ] › ℹ  info      Certbot Renewal Timer initialized
[IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
[Global   ] › ℹ  info      Backend PID 718 listening on port 48693 ...
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 14:14:51 [warn] 740#740: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 14:14:51 [warn] 740#740: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 14:14:51 [warn] 740#740: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 14:14:51 [notice] 740#740: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 15:14:52 [warn] 2097#2097: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 15:14:52 [warn] 2097#2097: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 15:14:52 [warn] 2097#2097: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 15:14:52 [notice] 2097#2097: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 16:14:52 [warn] 3439#3439: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 16:14:52 [warn] 3439#3439: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 16:14:52 [warn] 3439#3439: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 16:14:52 [notice] 3439#3439: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 17:14:52 [warn] 4797#4797: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 17:14:52 [warn] 4797#4797: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 17:14:52 [warn] 4797#4797: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 17:14:52 [notice] 4797#4797: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 18:14:51 [warn] 6146#6146: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 18:14:51 [warn] 6146#6146: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 18:14:51 [warn] 6146#6146: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 18:14:51 [notice] 6146#6146: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[SSL      ] › ℹ  info      Renewing TLS certs close to expiry...
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
[SSL      ] › ℹ  info      Renew Complete
2023/10/12 19:14:51 [warn] 7497#7497: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:14:51 [warn] 7497#7497: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:14:51 [warn] 7497#7497: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 19:14:51 [notice] 7497#7497: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/4.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/4.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/4.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/4.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/4.conf.err
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/4.conf.err"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 20:00:16 [warn] 8559#8559: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:16 [warn] 8559#8559: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:16 [warn] 8559#8559: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:16 [notice] 8559#8559: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/14/0)
Duplicate relation "access_list" in a relation expression. You should use "a.[b, c]" instead of "[a.b, a.c]". This will cause an error in objection 2.0
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/proxy_host/1.conf
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/12 20:00:57 [warn] 8722#8722: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:57 [warn] 8722#8722: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:57 [warn] 8722#8722: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/12 20:00:57 [notice] 8722#8722: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)

**Comment by Zoey2936** [comment]
I'm confused where exactly do you get the error "invalid port in 0.0.0.0.0.0.0.0:443"

**Comment by OpenK54** [comment]
I don't now. I didn't  set that ip. But when I change it manually in the config file tje problem persists.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Zoey2936** [reply]
no, I mean where you read this error log? in npmplus? in a program?

**Comment by OpenK54** [comment]
Npmplus docker log, extracted with portainer.

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Zoey2936** [reply]
I would like to help but I need the docker log with this error for that

**Comment by OpenK54** [comment]
I hope this helps:

2023/10/13 20:15:36 [emerg] 8913#8913: bind() to 0.0.0.0:443 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to [::]:443 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to 0.0.0.0:81 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to [::]:81 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: still could not bind()
/app/lib/utils.js:44
                                        reject(new Error(`Command '${cmd}' exited with code ${code}`));
                                               ^

Error: Command 'nginx -e stderr' exited with code 1
    at ChildProcess.<anonymous> (/app/lib/utils.js:44:13)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)

Node.js v20.5.0
----------------------------------------------------------
Let's Encrypt certificates cleanup - 2023/10/13 20:15:41
----------------------------------------------------------
0 file(s) kept.
0 file(s) deleted.
skipped '/data/etc/modsecurity/crs-setup.conf'
no NPM_CERT_ID set, using dummycerts for npm and default hosts.
Working on file: /data/nginx/custom/events.conf
Working on file: /data/nginx/custom/http.conf
Working on file: /data/nginx/custom/http_top.conf
Working on file: /data/nginx/custom/root.conf
Working on file: /data/nginx/custom/server_dead.conf
Working on file: /data/nginx/custom/server_proxy.conf
Working on file: /data/nginx/custom/server_redirect.conf
Working on file: /data/nginx/custom/server_stream.conf
Working on file: /data/nginx/custom/server_stream_tcp.conf
Working on file: /data/nginx/custom/server_stream_udp.conf
Working on file: /data/nginx/custom/stream.conf
Working on file: /data/nginx/default.conf
Working on file: /data/nginx/ip_ranges.conf
Success.
removed '/usr/local/nginx/logs/nginx.pid'

-------------------------------------
_   _ ____  __  __
| \ | |  _ \|  \/  |
|  \| | |_) | |\/| |
| |\  |  __/| |  | |
|_| \_|_|   |_|  |_|
-------------------------------------
Version:  2.10.4+36298f2
Date:     Fri Oct 13 20:15:42 UTC 2023
User:     root
PUID:     0
User ID:  0
PGID:     0
Group ID: 0
-------------------------------------

nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
[Global   ] › ℹ  info      Using Sqlite: /data/etc/npm/database.sqlite
[Migrate  ] › ℹ  info      Current database version: none
[IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
[IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
[SSL      ] › ℹ  info      Certbot Renewal Timer initialized
[IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
[Global   ] › ℹ  info      Backend PID 689 listening on port 48693 ...
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [notice] 709#709: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)
[Express  ] › ⚠  warning   invalid signature
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf.err
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf.err"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [notice] 1469#1469: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)



Von: Zoey ***@***.***>
Gesendet: Freitag, 13. Oktober 2023 08:48
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


I would like to help but I need the docker log with this error for that

**Reply by Zoey2936** [reply]
does npmplus always crash or only sometimes? and if only sometimes, when?

**Comment by OpenK54** [comment]
I always get an error (400) and the stream goes online. Here is an updated log after a second try with another stream:

2023/10/13 20:15:36 [emerg] 8913#8913: bind() to 0.0.0.0:443 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to [::]:443 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to 0.0.0.0:81 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: bind() to [::]:81 failed (98: Address in use)
2023/10/13 20:15:36 [emerg] 8913#8913: still could not bind()
/app/lib/utils.js:44
                                        reject(new Error(`Command '${cmd}' exited with code ${code}`));
                                               ^

Error: Command 'nginx -e stderr' exited with code 1
    at ChildProcess.<anonymous> (/app/lib/utils.js:44:13)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)

Node.js v20.5.0
----------------------------------------------------------
Let's Encrypt certificates cleanup - 2023/10/13 20:15:41
----------------------------------------------------------
0 file(s) kept.
0 file(s) deleted.
skipped '/data/etc/modsecurity/crs-setup.conf'
no NPM_CERT_ID set, using dummycerts for npm and default hosts.
Working on file: /data/nginx/custom/events.conf
Working on file: /data/nginx/custom/http.conf
Working on file: /data/nginx/custom/http_top.conf
Working on file: /data/nginx/custom/root.conf
Working on file: /data/nginx/custom/server_dead.conf
Working on file: /data/nginx/custom/server_proxy.conf
Working on file: /data/nginx/custom/server_redirect.conf
Working on file: /data/nginx/custom/server_stream.conf
Working on file: /data/nginx/custom/server_stream_tcp.conf
Working on file: /data/nginx/custom/server_stream_udp.conf
Working on file: /data/nginx/custom/stream.conf
Working on file: /data/nginx/default.conf
Working on file: /data/nginx/ip_ranges.conf
Success.
removed '/usr/local/nginx/logs/nginx.pid'

-------------------------------------
_   _ ____  __  __
| \ | |  _ \|  \/  |
|  \| | |_) | |\/| |
| |\  |  __/| |  | |
|_| \_|_|   |_|  |_|
-------------------------------------
Version:  2.10.4+36298f2
Date:     Fri Oct 13 20:15:42 UTC 2023
User:     root
PUID:     0
User ID:  0
PGID:     0
Group ID: 0
-------------------------------------

nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
[Global   ] › ℹ  info      Using Sqlite: /data/etc/npm/database.sqlite
[Migrate  ] › ℹ  info      Current database version: none
[IP Ranges] › ℹ  info      Fetching IP Ranges from online services...
[IP Ranges] › ℹ  info      Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v4
[IP Ranges] › ℹ  info      Fetching https://www.cloudflare.com/ips-v6
[SSL      ] › ℹ  info      Certbot Renewal Timer initialized
[IP Ranges] › ℹ  info      IP Ranges Renewal Timer initialized
[Global   ] › ℹ  info      Backend PID 689 listening on port 48693 ...
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [warn] 709#709: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:15:47 [notice] 709#709: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)
[Express  ] › ⚠  warning   invalid signature
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf"
}
[Nginx    ] › ⬤  debug     Deleting file: /data/nginx/stream/5.conf.err
[Nginx    ] › ⬤  debug     Could not delete file: {
  "errno": -2,
  "syscall": "unlink",
  "code": "ENOENT",
  "path": "/data/nginx/stream/5.conf.err"
}
[Nginx    ] › ℹ  info      Quitting Nginx
[Nginx    ] › ℹ  info      Starting Nginx in three seconds
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [warn] 1469#1469: "ssl_stapling" ignored, issuer certificate not found for certificate "/data/tls/dummycert.pem"
2023/10/13 20:46:08 [notice] 1469#1469: ModSecurity-nginx v1.0.3 (rules loaded inline/local/remote: 0/7/0)

Von: Zoey ***@***.***>
Gesendet: Freitag, 13. Oktober 2023 23:19
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


does npmplus always crash or only sometimes? and if only sometimes, when?

**Reply by Zoey2936** [reply]
so nginx crashes when you use the stream function?

**Comment by OpenK54** [comment]
In the backround it still seems to work, I can establish other connection with a proxy host at same time when the stream crashes; so it doesn’t crash entirely. It only gives the error and the stream doesn’t come online. And it is not possible to set the stream online by disablingand and enabling it.

________________________________

**Reply by Zoey2936** [reply]
sorry, no idea

**Comment by Zoey2936** [comment, marked as answer]
found the error, will be fixed in the next release

**Comment by OpenK54** [comment]
Great to hear that. Thank you so much for your efforts! Cheers, Rafael

Von: Zoey ***@***.***>
Gesendet: Freitag, 10. November 2023 06:18
An: ZoeyVid/NPMplus ***@***.***>
Cc: OpenK54 ***@***.***>; Author ***@***.***>
Betreff: Re: [ZoeyVid/NPMplus] Can't add a stream - stream always offline (Discussion #435)


found the error, will be fixed in the next release

-
